### PR TITLE
mbedtls: 2.16.3 -> 2.16.9

### LINKED ISSUE #99809: Vulnerability roundup 93: mbedtls-2.16.3: 1 advisory [5.5]
[search](https://search.nix.gsc.io/?q=mbedtls&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=mbedtls+in%3Apath&type=Code)

* [ ] [CVE-2020-16150](https://nvd.nist.gov/vuln/detail/CVE-2020-16150) CVSSv3=5.5 (nixos-20.03, nixos-20.09, nixos-unstable)

Scanned versions: nixos-20.03: 0d0660fde3b; nixos-20.09: ad3a5d5092e; nixos-unstable: 84d74ae9c9c.

Cc @fpletz

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/99809

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mbedtls/default.nix [hunk 1]
```diff
@@ -11,14 +11,13 @@
 
 stdenv.mkDerivation rec {
   pname = "mbedtls";
-  name = "mbedtls-${version}";
-  version = "2.16.3"; # nixpkgs-update: no auto update
+  version = "2.16.9"; # nixpkgs-update: no auto update
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mbedtls/default.nix:14]:
```suggestion
  version = "2.16.9";
```

Comment (dotlambda) [reply pkgs/development/libraries/mbedtls/default.nix:14]: There was a reason to disable auto-updates: 0ee02a9d42b5fe1825b0f7cee7a9986bb4ba975d

Comment (SuperSandro2000) [reply pkgs/development/libraries/mbedtls/default.nix:14]: Oh that is a magic comment. Did not know that.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy for me macos 10.14, linux x86_64. Only failures already existing.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111307` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>belcard</li>
<li>belle-sip</li>
<li>belr</li>
<li>bzrtp</li>
<li>dolphinEmuMaster</li>
<li>lime</li>
<li>ortp</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>gauche: log <a href="https://termbin.com/m2ly">https://termbin.com/m2ly</a></li>
  </ul>
</details>
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>gaucheBootstrap</li>
<li>haxe (haxe_3_4)</li>
<li>hxcpp (haxePackages.hxcpp)</li>
<li>haxePackages.hxcs</li>
<li>haxePackages.hxjava</li>
<li>haxePackages.hxnodejs_4</li>
<li>haxe_3_2</li>
<li>hiawatha</li>
<li>mbedtls</li>
<li>neko</li>
<li>shadowsocks-libev</li>
<li>yojimbo</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111307` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>obs-ndi: log was empty</li>
  </ul>
</details>
<details>
<summary>
36 packages built:
</summary>
<ul>
<li>bctoolbox</li>
<li>belcard</li>
<li>belle-sip</li>
<li>belr</li>
<li>bzrtp</li>
<li>dislocker</li>
<li>dolphinEmu</li>
<li>dolphinEmuMaster</li>
<li>gauche</li>
<li>gaucheBootstrap</li>
<li>haxe (haxe_3_4)</li>
<li>hxcpp (haxePackages.hxcpp)</li>
<li>haxePackages.hxcs</li>
<li>haxePackages.hxjava</li>
<li>haxePackages.hxnodejs_4</li>
<li>haxe_3_2</li>
<li>hiawatha</li>
<li>liblinphone</li>
<li>lime</li>
<li>linphone</li>
<li>lutris</li>
<li>lutris-free</li>
<li>mbedtls</li>
<li>mediastreamer</li>
<li>mediastreamer-openh264</li>
<li>msilbc</li>
<li>neko</li>
<li>obs-gstreamer</li>
<li>obs-move-transition</li>
<li>obs-studio</li>
<li>obs-v4l2sink</li>
<li>obs-wlrobs</li>
<li>ortp</li>
<li>shadowsocks-libev</li>
<li>trx</li>
<li>yojimbo</li>
</ul>
</details>

**Comment by dotlambda** [comment]
backport: https://github.com/NixOS/nixpkgs/pull/111401

**Pull request merged**